drivers/xen/gnttab.c

```diff
@@ -0,0 +1,336 @@
+/* SPDX-License-Identifier: MIT */
```

Comment:
I think there are some issues the License:

-   On the PR details, you said the work was based on Mini-OS. Looking at the COPYING file in the repo, the code is licensed as a BSD-2 clause rather than MIT.
- Some code seems to be inspired/copied from Linux. Which would imply there are some GPL-v2 code.

It is important to make sure that the license is correct and still compatible with Zephyr (I don't know much about Apache). Can you verify it?

Reply:
Hi Julien,
First of all I want to say a huge thank you for detailed review, it means a lot for me! I will process all of your comments as soon as possible.

About licensing: during writing a description of this PR I missed some important details - driver code was taken from Unikraft repository. Originally it was based on Mini-OS driver and this described in file description, so I said that driver in PR is Mini-OS based too (I will clarify this in description). Unikraft is MIT-licensed (probably it was allowed by copyright owner during Unikraft development) and MIT is compatible with Apache license, MIT code merging is possible in Zephyr (it was already done for some Xen public headers). So I think there is no problem with licensing from this side.

About a Linux side - no direct code copying was made, only implement GNTST_eagain handling in pretty same way, due to any tips missing about this situation in Xen. If it is a problem, I can try to completely re-write this.